demos/python_demos/face_recognition_demo/README.md

```diff
@@ -0,0 +1,187 @@
+the application deploys 4 models on the specified devices using OpenVINO
+library and runs them in asynchronous manner. The first model to be used
+is the face detection model, which is followed by two independent models
+for head pose prediction and face keypoints prediction. The last step in
+frame processing is done by face recognition model, which uses keypoints
+to align the faces and the face gallery to match faces found on a video
+frame with the ones in the gallery. Then, the processing results are
+visualized and displayed on the screen or written to the output file.
+
+### Creating a gallery for face recognition
+
+To recognize faces the application uses a face database, or a gallery.
+The gallery is a folder with images of persons. Each image in the gallery can
+be of arbitrary size and should contain one or more frontally-oriented faces
+with decent quality. The application can use face detector during the gallery
+build, this is controlled by "--run_detector" flag. This allows gallery images
+to contain more than one face image and not to be tightly cropped.
+
+Image file name is used as a person name during the visualization.
+Use the following name convention: `person_N_name.png` or `person_N_name.jpg`.
+
+### Installation and dependencies
+
+The demo depends on:
+- OpenVINO library (R5)
+- Python (any of 2.7+ or 3.4+, which is supported by OpenVINO)
+- NumPy (>=1.11.0)
+- SciPy (>=1.1.0)
+- OpenCV (>=3.4.0)
+
+To install all the required Python modules you can use:
+
+``` sh
+pip install -r requirements.txt
+```
+
+### Running the demo:
+
+Running the application with the `-h` option or without
+any arguments yields the following message:
+
+``` sh
+./face_recognition_demo.py -h
+
+usage: face_recognition_demo.py [-h] [-i PATH] [-o PATH] [-no_show] [-tl]
+                                [-cw CROP_WIDTH] [-ch CROP_HEIGHT] -fg PATH
+                                [--run_detector] -m_fd PATH -m_lm PATH -m_reid
+                                PATH -m_hp PATH
+                                [-d_fd {CPU,GPU,FPGA,MYRIAD,HETERO}]
+                                [-d_lm {CPU,GPU,FPGA,MYRIAD,HETERO}]
+                                [-d_reid {CPU,GPU,FPGA,MYRIAD,HETERO}]
+                                [-d_hp {CPU,GPU,FPGA,MYRIAD,HETERO}] [-l PATH]
+                                [-c PATH] [-v] [-pc] [-t_fd [0..1]]
+                                [-t_id [0..1]] [-exp_r_fd NUMBER]
+
+optional arguments:
+  -h, --help            show this help message and exit
+
+General:
+  -i PATH, --input PATH
+                        (optional) Path to the input video ('cam' for the
+                        camera, default)
+  -o PATH, --output PATH
+                        (optional) Path to save the output video to
+  -no_show              (optional) Do not display output
+  -tl, --timelapse      (optional) Auto-pause after each frame
+  -cw CROP_WIDTH, --crop_width CROP_WIDTH
```

Comment:
These parameters are optional - if they are not used, the full input frame is processed.

Reply:
Well, `--run_detector` has the wording "otherwise use full images.". Most other optional options (pun!) also tell you what happens if you omit it. These do not.